File: go-client-sdk/pkg/models/shared/dealcreate.go
package shared

type DealCreate struct {
	AccountID   *string                `json:"accountId,omitempty"`
	Additional  map[string]interface{} `json:"additional,omitempty"`
	Amount      *float64               `json:"amount,omitempty"`
	CloseDate   string                 `json:"closeDate"`
	Name        string                 `json:"name"`
	Probability *string                `json:"probability,omitempty"`
	Stage       *string                `json:"stage,omitempty"`
}

File: typescript-client-sdk/src/sdk/models/shared/dealcreate.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class DealCreate extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=accountId" })
  accountId?: string;

  @SpeakeasyMetadata({ data: "json, name=additional" })
  additional?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=amount" })
  amount?: number;

  @SpeakeasyMetadata({ data: "json, name=closeDate" })
  closeDate: string;

  @SpeakeasyMetadata({ data: "json, name=name" })
  name: string;

  @SpeakeasyMetadata({ data: "json, name=probability" })
  probability?: string;

  @SpeakeasyMetadata({ data: "json, name=stage" })
  stage?: string;
}
